init.lua

```diff
@@ -0,0 +1,113 @@
+local INVIS_DURATION = 10
+local BUBBLE_INTERVAL = 0.2
+local POTION_COOLDOWN = 30 
+local invisible_players = {}
+local last_drink_time = {}
+
+minetest.register_craftitem("invis_potion:potion", {
+    description = "Invisible Potion",
+    inventory_image = "invis_potion.png",
+    on_use = function(itemstack, user, pointed_thing)
+        if not user or not user:is_player() then return end
+        local name = user:get_player_name()
+
+        local now = os.time()
+        if last_drink_time[name] and now - last_drink_time[name] < POTION_COOLDOWN then
+            local remaining = POTION_COOLDOWN - (now - last_drink_time[name])
+            minetest.chat_send_player(name, "Potion is on cooldown! Wait " .. remaining .. " seconds.")
+            return itemstack
+        end
+
+        user:set_properties({
+            visual_size = {x = 0, y = 0},
+            makes_footstep_sound = false
+        })
+        user:set_observers({})
+        invisible_players[name] = {
+            time_left = INVIS_DURATION,
+            bubble_timer = 0
+        }
+
+        last_drink_time[name] = now
+        itemstack:take_item()
+
+        minetest.chat_send_player(name, "You are invisible now!")
+
+        return itemstack
+    end,
+})
+
+minetest.register_globalstep(function(dtime)
+    for name, data in pairs(invisible_players) do
+        local player = minetest.get_player_by_name(name)
+        if not player then
+            invisible_players[name] = nil
+        else
+            data.time_left = data.time_left - dtime
+            data.bubble_timer = data.bubble_timer + dtime
+
+            if data.bubble_timer >= BUBBLE_INTERVAL then
+                data.bubble_timer = 0
+                local pos = player:get_pos()
+                local offset = {
+                    x = math.random(-5, 5) * 0.1,
+                    y = math.random(0, 15) * 0.1,
+                    z = math.random(-5, 5) * 0.1
+                }
+                minetest.add_particle({
+                    pos = {x=pos.x+offset.x, y=pos.y+1+offset.y, z=pos.z+offset.z},
```

Comment:
```suggestion
                core.add_particle({
                    pos = pos + vector.new(math.random(-0.5, 0.5), math.random(1, 2.5), math.random(-0.5, 0.5)),
```
you don't need to manually specify the x y and z axes if you have _vectors_